rust/syntect-plugin/src/main.rs

```diff
@@ -192,9 +193,13 @@ impl<'a> Syntect<'a> {
     }
 
     /// Wipes any existing state and starts highlighting with `syntax`.
-    fn do_highlighting(&mut self, view: &mut MyView) {
+    fn do_highlighting(&mut self, view: &mut MyView, lang: Option<&LanguageId>) {
```

Comment:
not sure if this signature is clear enough

Reply:
I'm slightly confused by why we need to add this param; it seems to make things more complicated. By associating the syntax with the view object, we can make sure we're using the correct syntax for the correct view. I'll point out where I see this being a problem, below